### PR TITLE
feat(docker): make sure `docker stop` stops all containers

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch makes sure we stop all containers of all Playwright
versions.

### CODE REVIEW

**Review by yury-s — APPROVED** [review]

File: packages/playwright-test/src/docker/docker.ts [hunk 1]
```diff
@@ -172,15 +182,22 @@ export async function ensureContainerOrDie(): Promise<ContainerInfo> {
...
-export async function stopContainer() {
-  const containerId = await findRunningDockerContainerId();
-  if (!containerId)
+export async function stopAllContainers() {
+  const allContainers: Container[] | undefined = await getJSON('/containers/json');
+  if (!allContainers?.length)
     return;
-  await Promise.all([
+  const allImages: DockerImage[] | undefined = await getJSON('/images/json');
+  // Get all docker images from current and past Playwright versions.
```
Comment (mxschmitt) [inline packages/playwright-test/src/docker/docker.ts:190]:
```suggestion
  // Get all docker containers from current and past Playwright versions.
```

**Review by dgozman — APPROVED** [review]

File: packages/playwright-test/src/docker/docker.ts [hunk 2]
```diff
@@ -172,15 +182,22 @@ export async function ensureContainerOrDie(): Promise<ContainerInfo> {
...
+  const dockerImageIds = new Set(allImages?.filter(image => image.RepoTags?.some(repoTag => VRT_IMAGE_NAME_REGEX.test(repoTag))).map(image => image.Id) ?? []);
```
Comment (mxschmitt) [inline packages/playwright-test/src/docker/docker.ts:191]: Would be nice to only stop containers launched by Playwright itself. So add a meta label on run and filter by it.

Comment (aslushnikov) [reply packages/playwright-test/src/docker/docker.ts:191]: Agreed. Closed this in favor of https://github.com/microsoft/playwright/pull/17316

**Comment by aslushnikov** [comment]
closing in favor of https://github.com/microsoft/playwright/pull/17316

**Pull request closed without merging**